src/nwb2bids/_converters/_dataset_converter.py

```diff
@@ -53,6 +66,20 @@ def from_nwb_directory(
         dataset_converter = cls(session_converters=session_converters, dataset_description=dataset_description)
         return dataset_converter
 
+    # TODO: remove
+    @classmethod
+    @pydantic.validate_call
+    def from_nwb_directory(
+        cls, nwb_directory: pydantic.DirectoryPath, additional_metadata_file_path: pydantic.FilePath | None = None
+    ) -> typing_extensions.Self:
+        message = "The method `.from_nwb_directory` is deprecated. Please use `.from_nwb` instead."
```

Comment:
IMHO at this age of development I would not bother about migrations/deprecations , especially at python API level